### ISSUE TITLE
Test failure: GC\\Features\\HeapExpansion\\bestfit-finalize\\bestfit-finalize.cmd

### ISSUE THREAD

**Issue by v-haren** [dotnet/runtime#45091]
failed in job: [ runtime-coreclr gcstress-extra 20201121.1 ](https://dev.azure.com/dnceng/public/_build/results?buildId=895622&view=ms.vss-test-web.build-test-results-tab&runId=28622992&resultId=109960&paneView=debug)

failed tests: 
GC\\Features\\HeapExpansion\\bestfit-finalize\\bestfit-finalize.cmd
GC\\Features\\HeapExpansion\\bestfit_1\\bestfit_1.cmd
GC\\Scenarios\\ServerModel\\servermodel\\servermodel.cmd
GC\\Features\\HeapExpansion\\bestfit\\bestfit.cmd

Error message
~~~
Assert failure(PID 2396 [0x0000095c], Thread: 4704 [0x1260]): pMT

CORECLR! WKS::GCHeap::ValidateObjectMember + 0x160 (0x00007ffe19696f70)
CORECLR! Object::ValidateInner + 0x311 (0x00007ffe192173d1)
CORECLR! Object::Validate + 0x118 (0x00007ffe19217078)
CORECLR! WKS::gc_heap::compact_plug + 0x1E0 (0x00007ffe196a2560)
CORECLR! WKS::gc_heap::compact_in_brick + 0x123 (0x00007ffe196a1a5f)
CORECLR! WKS::gc_heap::compact_phase + 0x324 (0x00007ffe196a22ac)
CORECLR! WKS::gc_heap::plan_phase + 0x20BC (0x00007ffe196b594c)
CORECLR! WKS::gc_heap::gc1 + 0x276 (0x00007ffe196a9456)
CORECLR! WKS::gc_heap::garbage_collect + 0x4FC (0x00007ffe196a912c)
CORECLR! WKS::GCHeap::GarbageCollectGeneration + 0x292 (0x00007ffe196943de)
    File: F:\workspace\_work\1\s\src\coreclr\src\gc\gc.cpp Line: 36783
    Image: C:\h\w\AE700977\p\CoreRun.exe


Return code:      1
Raw output file:      C:\h\w\AE700977\w\AD710958\e\GC\Features\Reports\GC.Features\HeapExpansion\bestfit-finalize\bestfit-finalize.output.txt
Raw output:
BEGIN EXECUTION
 "C:\h\w\AE700977\p\corerun.exe" bestfit-finalize.dll 
Using defaults: 1 1200 100000
Using random seed: -186711402
Expected: 100
Actual: -1073740286
END EXECUTION - FAILED
FAILED
Test Harness Exitcode is : 1
To run the test:
> set CORE_ROOT=C:\h\w\AE700977\p
> C:\h\w\AE700977\w\AD710958\e\GC\Features\HeapExpansion\bestfit-finalize\bestfit-finalize.cmd
Expected: True
Actual:   False


Stack trace
   at GC_Features._HeapExpansion_bestfit_finalize_bestfit_finalize_._HeapExpansion_bestfit_finalize_bestfit_finalize_cmd() in F:\workspace\_work\1\s\artifacts\tests\coreclr\windows.x64.Checked\TestWrappers\GC.Features\GC.Features.XUnitWrapper.cs:line 1009
~~~

**Comment by mangod9** [comment]
Looks like a heap corruption.

**Comment by Maoni0** [comment]
@v-haren pardon my ignorance - are these tests run on every PR? if this is tests that don't run in every PR and only run via other ways, I'd like to know how you run them so we can run themselves ourselves locally.

**Comment by PeterSolMS** [comment]
I will try to repro and debug this failure.

**Comment by mangod9** [comment]
Thanks @PeterSolMS

**Comment by PeterSolMS** [comment]
As it turns out, this readily repros.

The root cause is an overeager call to Validate() - the parameter bDeep defaults to TRUE, which means that we'll validate pointers to other objects as well if heap verify is on (more precisely, if the HEAPVERIFY_GC bit is set).

There is an easy one line fix for it - see PR #45360

**Comment by PeterSolMS** [comment]
New PR #45409
